src/main/java/bernard/core/Parser.java

```
@@ -0,0 +1,85 @@
+            } else if (commandArgs[0].equals("list")) {
+                taskList.listTasks();
+            } else if (commandArgs[0].equals("mark")) {
+                if (commandArgs.length == 1) {
+                    throw new BernardException("No task specified!");
+                }
+                try {
+                    int index = Integer.parseInt(commandArgs[1]) - 1;
+                    taskList.markTask(index);
+                } catch (NumberFormatException exception) {
+                    throw new BernardException("Invalid task index!");
+                }
+            } else if (commandArgs[0].equals("unmark")) {
+                if (commandArgs.length == 1) {
+                    throw new BernardException("No task specified!");
+                }
+                try {
+                    int index = Integer.parseInt(commandArgs[1]) - 1;
+                    taskList.unmarkTask(index);
+                } catch (NumberFormatException exception) {
+                    throw new BernardException("Invalid task index!");
+                }
+            } else if (commandArgs[0].equals("delete")) {
+                if (commandArgs.length == 1) {
+                    throw new BernardException("No task specified!");
+                }
+                try {
+                    int index = Integer.parseInt(commandArgs[1]) - 1;
+                    taskList.deleteTask(index);
+                } catch (NumberFormatException exception) {
+                    throw new BernardException("Invalid task index!");
+                }
+            } else if (commandArgs[0].equals("find")) {
+                if (commandArgs.length == 1) {
+                    throw new BernardException("No keyword specified!");
+                }
+                String keyword = String.join(" ",
+                        java.util.Arrays.copyOfRange(commandArgs, 1, commandArgs.length)).trim();
+                taskList.listMatchingTasks(keyword);
+            } else {
```

Any reasons why invalid commands will go through addTask()? Would it be better to just validate the know commands and throw exceptions for those that are unknown?